frontend/src/pages/registration/RegistrationPageView.jsx

```diff
@@ -0,0 +1,94 @@
+import { Button, styled, TextField } from '@mui/material'
+import React, { useState } from 'react'
+import classes from './registrationpage.module.scss'
+
+const ConfirmButton = styled(Button)({
+    display: 'flex',
+    fontSize: '16px',
+    padding: '10px 60px',
+    boxShadow:
+        'rgba(0, 0, 0, 0) 0px 10px 38px, rgba(0, 0, 0, 0.22) 0px 15px 12px',
+    cursor: 'pointer',
+    transition: 'all 0.1s',
+    margin: 'auto',
+})
+const InputField = styled(TextField)({
+    minWidth: '300px',
+    margin: '20px',
+})
+
+const RegistrationFormView = ({ submitForm, onInputChange, value, errors }) => {
+    const [error, setErrors] = useState({})
+    const [dataIsCollected, setDataIsCollected] = useState(false)
+
+    const formConfirm = async (event) => {
+        event.preventDefault()
+        setErrors(errors)
+        setDataIsCollected(true)
+        // if user data is collected and no errors
+        // set the subnitForm to true
+        // passing this value to RegistrationFormController later
+        // to show the Account Creat form
+        if (Object.keys(errors).length === 0 && dataIsCollected) {
+            submitForm(true)
+            console.log(value)
+        }
```

Comment:
When you setDataIsCollected(true), it doesn't get set instantly because it is put in the react queue. So sometimes, the code will reach line 32 and dataIsCollected will be false and the form won't submit properly causing the user to have to click twice.

You can just remove the whole dataIsCollected state and this will work 

Also, the function also doesn't need to be async, and your submitForm function accepts no argument so it can just be submitForm()

e.g. with all changes:
```js
const formConfirm = (event) => {
  event.preventDefault()
  setErrors(errors)

  if (Object.keys(errors).length === 0) {
    submitForm()
  }
}
```